### DISCUSSION TITLE
Error: Type 'UnmodifiableUint8ListView' not found.

### DISCUSSION THREAD

**Discussion by MilanRibadiya** [codemagic-ci-cd/codemagic-docs#2678] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I getting this type of error at archiving time : 

ios/Runner/AppDelegate.swift uses the deprecated @UIApplicationMain attribute, updating.
Automatically signing iOS for device deployment using specified development team in Xcode project: VU84QVU7F7
Running pod install...                                             46.6s
Running Xcode build...                                          
Xcode archive done.                                         44.6s
Failed to build iOS app
Error (Xcode): ../.pub-cache/hosted/pub.dev/win32-5.5.0/lib/src/guid.dart:32:9: Error: Type 'UnmodifiableUint8ListView' not found.

Encountered error while archiving for device.

Build failed :|
Failed to build for iOS

### Expected results

Creating archive and upload to app store.

### Actual results

I getting error at archiving : 

ios/Runner/AppDelegate.swift uses the deprecated @UIApplicationMain attribute, updating.
Automatically signing iOS for device deployment using specified development team in Xcode project: VU84QVU7F7
Running pod install...                                             46.6s
Running Xcode build...                                          
Xcode archive done.                                         44.6s
Failed to build iOS app
Error (Xcode): ../.pub-cache/hosted/pub.dev/win32-5.5.0/lib/src/guid.dart:32:9: Error: Type 'UnmodifiableUint8ListView' not found.

Encountered error while archiving for device.

### Build id (optional)

66b7b85b16035adf8e2fd58a

**Comment by zelle7** [comment]
I face an similar issue on an android build might be a duplicate?


````
../.pub-cache/hosted/pub.dev/win32-5.5.0/lib/src/guid.dart:52:17: Error: Method not found: 'UnmodifiableUint8ListView'.
    return Guid(UnmodifiableUint8ListView(guid));
                ^^^^^^^^^^^^^^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/win32-5.5.0/lib/src/guid.dart:56:31: Error: Method not found: 'UnmodifiableUint8ListView'.
  factory Guid.zero() => Guid(UnmodifiableUint8ListView(Uint8List(16)));
                              ^^^^^^^^^^^^^^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/win32-5.5.0/lib/src/guid.dart:100:17: Error: Method not found: 'UnmodifiableUint8ListView'.
    return Guid(UnmodifiableUint8ListView(Uint8List.fromList(guidAsBytes)));
                ^^^^^^^^^^^^^^^^^^^^^^^^^
Target kernel_snapshot_program failed: Exception


FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script '/Users/builder/programs/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1687

* What went wrong:
Execution failed for task ':app:compileFlutterBuildRelease'.
> Process 'command '/Users/builder/programs/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 5s
Running Gradle task 'bundleRelease'...                             66.3s
Gradle task bundleRelease failed with exit code 1


Build failed :|
Failed to build for Android
````

## Build Id

66b8d807ae55a87327a8b7da

**Reply by dtrdic** [reply]
Hi @zelle7, you as well, try to lower the flutter version from `channel Stable` to `3.22.3`

**Reply by zelle7** [reply]
@dtrdic thx I did 

````
flutter pub cache clean
flutter channel stable
flutter pub get
````

This seemed to help. Thx!

**Comment by dtrdic** [comment, marked as answer]
Hi @MilanRibadiya please try lowering the flutter version from flutter `channel Stable` to `3.22.3`. It seems that this error is occurring in the stable version that was released a few days ago by Flutter. Check also this ongoing [discussion](https://github.com/cbl-dart/cbl-dart/issues/598) for reference 
![image](https://github.com/user-attachments/assets/ea276ea7-4fa0-4a1d-9a49-877bd0e09241)

**Comment by MilanRibadiya** [comment]
@dtrdic It's working for me. Thanks for correct answer.

**Comment by al-batol** [comment]
```
flutter pub cache clean
flutter pub upgrade win32
flutter channel stable
flutter pub get
flutter pub outdated
flutter pub upgrade --major-versions
flutter run
```

**Reply by iamvalenciia** [reply]
works, thank you!

**Reply by clemens-unterkofler** [reply]
had to do a local `flutter upgrade` first before it would update the dependencies. it removed the offending win32 dependency.
also:
WIN? 32? at this time of the year? at this latitude? on a mac build system? building for ios/android?

**Reply by cloud-prg** [reply]
resolved! useful~

**Reply by Shibil-77** [reply]
resolved

**Reply by imujtaba8488** [reply]
`flutter pub upgrade win32`

**Reply by XavierSoh** [reply]
Works Fine !!!

**Reply by MithSeang** [reply]
I have this problem but I follow all these step its not working :((( idk whyy

**Reply by MapIHS** [reply]
resolved

**Reply by sreerag2460** [reply]
> ```
> flutter pub cache clean
> flutter pub upgrade win32
> flutter channel stable
> flutter pub get
> flutter pub outdated
> flutter pub upgrade --major-versions
> flutter run
> ```

  
  
  Thanks , it works

**Reply by organizedslop** [reply]
This worked for me as well, thank you!

**Reply by qq1100** [reply]
非常Nice
